streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -885,6 +885,57 @@ public void shouldNotCommitOnHandleAssignmentIfOnlyStandbyTaskClosed() {
         assertThat(task10.commitPrepared, is(false));
     }
 
+    @Test
+    public void shouldCleanupAnyTasksClosedAsDirtyAfterCommitException() {
```

Comment:
Verified that this test does fail on `verify(changelogReader)` without this fix

Reply:
Great to see this test case !

Reply:
It's 500% easier to write tests for this thanks to the refactoring work @guozhangwang did for `TaskManager` and `TaskManagerTest` 🙂